### PR TITLE
Policy for typography

### LINKED ISSUE #2599: Typography guideline
With the recent typography tuning, I think it would be useful to document somewhere what we have done. Unfortunately, my English is quite bad. But I have a draft that maybe someone could review:

# Typography

This style uses the font “Noto” (and the font “Unifont” as last resort fall-back) for a world-wide covering of scripts. The font size should be ≥ 10 (legibility).

### Multi-line labels

Additional to text-size  we have to set text-wrap-width and text-line-spacing, For both, the absolute value is meaningless, it should be interpreted relative to the font size (em):
```
text-size: 10;
text-wrap-width: 30; // 3.0 em
text-line-spacing: -1.5; // -0.15 em
```
If text-size increases on higher zoom levels, the other parameters must be adjusted also:
```
text-size: 12;
text-wrap-width: 36; // 3.0 em
text-line-spacing: -1.8; // -0.15 em
```
Usually, with higher zoom levels we increase the line length (text-wrap-width in em). Following an old typography convention, we use narrow line spacing for short lines and wider line spacing for longer lines.
```
text-size: 15;
text-wrap-width: 75; // 5.0 em
text-line-spacing: -0.75; // -0.05 em
```
Noto’s line spacing is rather large to allow also tall scripts like Myanmar to be rendered without collisions. But the line spacing is too large for cartographic usage, therefor we reduce the line spacing. Currently, the line spacing ranges from -0.15 em to -0.05 em. (Even at -0.15 em, collisions are seldom and even then the text stays legible).

### PR DESCRIPTION
**Pull request by sommerluk**
Resolves #2599

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -133,6 +133,42 @@ The project's goals and design philsophy are different from other projects, but
 * [Maki Icons Design Guidelines](https://www.mapbox.com/maki-icons/guidelines/)
 * [GNOME Icon Design Guildelines](https://developer.gnome.org/hig/stable/icons-and-artwork.html.en)
 
+## Typography
+
+This style uses the font "Noto" for a world-wide covering of scripts. The font
```
Comment (nebulon42) [inline CONTRIBUTING.md]: covering should be coverage

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -133,6 +133,42 @@ The project's goals and design philsophy are different from other projects, but
...
+
+Additional to text-size we have to set text-wrap-width and text-line-spacing.
+For both, the absolute value is meaningless; it should be interpreted relative
+to the font size (em):
+```
+text-size: 10;
+text-wrap-width: 30; // 3.0 em
+text-line-spacing: -1.5; // -0.15 em
+```
+If text-size increases on higher zoom levels, the other parameters must also be
```
Comment (nebulon42) [inline CONTRIBUTING.md]: I think the comma can be left out. "must also be" could also read "also have to be"

**Comment by nebulon42** [comment]
@sommerluk Could you please amend "resolves #2599" to the commit message so that the issue is automatically closed?

**Comment by sommerluk** [comment]
Thanks. Ready.

**Comment by sommerluk** [comment]
> @sommerluk Could you please amend "resolves #2599" to the commit message so that the issue is automatically closed?

I had “Resolves #2599” in the initial comment. Isn’t that now recognized by Github also?

**Comment by HolgerJeromin** [comment]
> I had “Resolves #2599” in the initial comment. Isn’t that now recognized by Github also?

Yes it is.

**Comment by nebulon42** [comment]
Since your proposed text was available in #2599 since some time I'm merging this right away.

**Pull request merged**